lib/ldap.js

```diff
@@ -28,7 +28,7 @@ function initializeConnection () {
   });
 
   connection.heartbeat = function (callback) {
-    connection.search('', '(objectclass=*)', function (err, res) {
+    connection.search('', '(&(objectclass=user)(|(sAMAccountName=foo)(UserPrincipalName=foo)))', function (err, res) {
```

Comment:
Two suggestions:
- Constraint the search under configured base DN, will make the search faster
- Make heartbeat search query configurable. We can have the above query in PR as default

Reply:
please go ahead with your suggestions, we have a customer waiting for this change.